### ISSUE TITLE
Prevent duplicate spinners in legendary spinner list

### ISSUE THREAD

**Issue by mayppong** [mayppong/gyro#20]
Since we're currently only add spinner to the list but not checking for duplicate, the same spinner could show up multiple times.

This one will likely depends on #32 to be implemented first otherwise we won't be able to identify one spinner from another.

**Comment by mayppong** [comment]
For now I am using the spinner pid as a unique identifier and the logic is in place to check for that. I think we could close this ticket as a result even if #32 isn't completed yet. Once it's done, this ticket is no longer relevant anyway so there's no point in leaving it open.
